apollo-ios/Sources/Apollo/GraphQLExecutor.swift

```diff
@@ -265,27 +265,76 @@ public final class GraphQLExecutor<Source: GraphQLExecutionSource> {
     info: ObjectExecutionInfo,
     accumulator: Accumulator
   ) -> PossiblyDeferred<Accumulator.ObjectResult> {
+    let fieldEntries: [PossiblyDeferred<Accumulator.FieldEntry?>] = execute(
+      selections: selections,
+      on: object,
+      info: info,
+      accumulator: accumulator
+    )
+
+    return compactLazilyEvaluateAll(fieldEntries).map {
+      try accumulator.accept(fieldEntries: $0, info: info)
+    }
+  }
+
+  private func execute<Accumulator: GraphQLResultAccumulator>(
+    selections: [Selection],
+    on object: Source.RawObjectData,
+    info: ObjectExecutionInfo,
+    accumulator: Accumulator
+  ) -> [PossiblyDeferred<Accumulator.FieldEntry?>] {
     do {
       let groupedFields = try groupFields(selections, on: object, info: info)
       info.fulfilledFragments = groupedFields.fulfilledFragments
-      info.deferredFragments = groupedFields.deferredFragments
+      info.deferredFragments = []
 
       var fieldEntries: [PossiblyDeferred<Accumulator.FieldEntry?>] = []
       fieldEntries.reserveCapacity(groupedFields.count)
 
-      for (_, fields) in groupedFields {
-        let fieldEntry = execute(fields: fields,
-                                 on: object,
-                                 accumulator: accumulator)
+      for (_, fields) in groupedFields.fieldInfoList {
+        let fieldEntry = execute(
+          fields: fields,
+          on: object,
+          accumulator: accumulator)
         fieldEntries.append(fieldEntry)
       }
-      
-      return compactLazilyEvaluateAll(fieldEntries).map {
-        try accumulator.accept(fieldEntries: $0, info: info)
+
+      if executionSource.shouldAttemptDeferredFragmentExecution {
+        for deferredFragment in groupedFields.deferredFragments {
+          guard let fragmentType = groupedFields.cachedFragmentIdentifiers[deferredFragment] else {
+            info.deferredFragments.insert(deferredFragment)
```

Comment:
In the case that the fragment type cannot be found in the cache I think it's best that it remains 'deferred'; it's not going to break execution.